examples/material_gallery/lib/demo/weathers_demo.dart

```diff
@@ -0,0 +1,495 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui;
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_sprites/flutter_sprites.dart';
+
+ImageMap _images;
+SpriteSheet _sprites;
+
+enum WeatherType {
+  sun,
+  rain,
+  snow
+}
+
+class WeathersDemo extends StatefulComponent {
+  _WeathersDemoState createState() => new _WeathersDemoState();
+}
+
+class _WeathersDemoState extends State<WeathersDemo> {
+
+  Future _loadAssets(AssetBundle bundle) async {
+    _images = new ImageMap(bundle);
+    await _images.load([
+      'packages/flutter_gallery_assets/clouds-0.png',
+      'packages/flutter_gallery_assets/clouds-1.png',
+      'packages/flutter_gallery_assets/ray.png',
+      'packages/flutter_gallery_assets/sun.png',
+      'packages/flutter_gallery_assets/weathersprites.json',
+      'packages/flutter_gallery_assets/weathersprites.png',
+      'packages/flutter_gallery_assets/icon-sun.png',
+      'packages/flutter_gallery_assets/icon-rain.png',
+      'packages/flutter_gallery_assets/icon-snow.png'
+    ]);
+
+    String json = await DefaultAssetBundle.of(context).loadString('packages/flutter_gallery_assets/weathersprites.json');
+    _sprites = new SpriteSheet(_images['packages/flutter_gallery_assets/weathersprites.png'], json);
+  }
+
+  void initState() {
+    super.initState();
+
+    AssetBundle bundle = DefaultAssetBundle.of(context);
+    _loadAssets(bundle).then((_) {
+      setState(() {
+        assetsLoaded = true;
+        weatherWorld = new WeatherWorld();
+      });
+    });
+  }
+
+  bool assetsLoaded = false;
+
+  WeatherWorld weatherWorld;
+
+  Widget build(BuildContext context) {
+    if (!assetsLoaded)
+      return new Text("Loading...");
```

Comment:
We should probably just use a solid colour background rather than blocking the entire UI on the assets loading.